deepcell_tracking/utils.py

```diff
@@ -505,28 +508,32 @@ def is_valid_lineage(y, lineage):
         frames = list(y_index)
         if frames != cell_lineage['frames']:
             warnings.warn('Cell {} has invalid frames'.format(cell_label))
-            return False
+            is_valid = False
+            continue  # no need to test further
 
         last_parent_frame = cell_lineage['frames'][-1]
 
         for daughter in cell_lineage['daughters']:
-            if daughter not in all_cells or daughter not in lineage:
+            if daughter not in lineage:
```

Comment:
Does this hold in the case that a cell has random number (either 0 or not in `np.unique(y)` as an entry in `cell_lineage['daughters']`

Reply:
This is a simple check to make sure that the daughter cells exist in the lineage. Other checks will happen when we hit the daughter ID in the outer for loop (for cellID in lineage).